src/main/java/org/truffleruby/options/ByteStringOptionDescription.java

```diff
@@ -47,4 +50,11 @@ public String toString(Object value) {
         }
     }
 
+    private static final OptionType<byte[]> OPTION_TYPE = new OptionType<>("byte[]", new byte[]{}, (Function<String, byte[]>) text -> { throw new UnsupportedOperationException(); });
```

Comment:
Extracting the throw in a properly-typed function should help.
But why not use checkValue?